lib/ansible/executor/task_executor.py

```diff
@@ -410,6 +410,20 @@ def _run_loop(self, items):
             results.append(res)
             del task_vars[loop_var]
 
+            # clear 'connection related' plugin variables for next iteration
+            if self._connection:
+                clear_plugins = {
+                    'connection': self._connection._load_name,
+                    'shell': self._connection._shell._load_name
+                }
+                if self._play_context.become:
```

Comment:
connections can load their own become plugin, so it might be better to check on the connection itself

Reply:
TIL, fixed, now the change is "`play_context` free".